### PR TITLE
[VTP-1817] Delete CDA token on "close" github action

### PR DESCRIPTION
**Pull request by slash3b**
This PR adds a couple of fixes:
- CDA token is deleted when PR either closed or merged
- ephemeral environment deleted not only when PR is merged but when PR is closed as well

Test scenario executed manually:
- open up PR, new env and token are created ✔️ 
- close PR, env and token are delete ✔️ 
- reopen PR, new env and new token are created ✔️ 
- merge PR, new env and new token are deleted ✔️ 

Dependent PR(s):
- https://github.com/the-infatuation/contentful-action/pull/10

Example of working test scenario:  

Screenshot 2023-10-24 at 15 53 29

### CODE REVIEW

**Review by steenzout — DISMISSED** [review]
LGTM!

Above and beyond! Thank you! 💪🏻 👏🏻 👏🏻 👏🏻

File: src/action.ts [hunk 1]
```diff
@@ -231,14 +231,38 @@ export const runAction = async (space): Promise<void> => {
...
 
+  // "closed" action happens on PR close and PR merge
+  const githubAction = github.context.payload.action
+
+  // If CDA token is created and we want to purge ephemeral env upon close/merge
+  // then ephemeral token should be deleted as well
+  if ( 
+    DELETE_FEATURE && 
+    CREATE_CDA_TOKEN && 
+    githubAction == "closed"
```
Comment (kayakyakr) [inline src/action.ts:242]: make sure you're using `===` instead of `==` across any js/ts code.

**Review by kayakyakr — COMMENTED** [review]
@steenzout this would be another candidate for splitting up, if we ever wanted to cycle back to this and separate action duties. EG: this would be a separate action to just delete ephemeral keys and/or environments that's triggered in a workflow only on close.

But for now, this is fine.

File: src/action.ts [hunk 2]
```diff
@@ -231,14 +231,38 @@ export const runAction = async (space): Promise<void> => {
...
+  if ( 
+    DELETE_FEATURE && 
+    CREATE_CDA_TOKEN && 
+    githubAction === "closed" 
+  ) {
+    const { items: keys } = await space.getApiKeys();
+    keys.map((key) => {
+      if (key.name === tokenKeyName) {
+        try {
+          key.delete()
```
Comment (kayakyakr) [inline src/action.ts:248]: Does this need to be `await`ed?

**Comment by kayakyakr** [comment]
Should we remove/disable the migration example workflow? It's going to fail and since this is a public repo, we don't really want to connect it up to a contentful repo that actually works, I don't think.

**Comment by steenzout** [comment]
> Should we remove/disable the migration example workflow? It's going to fail and since this is a public repo, we don't really want to connect it up to a contentful repo that actually works, I don't think.

idk.

I've disabled the workflow from command-line (see #12) so it shouldn't run anymore unless someone enables the workflow again.

File: src/action.ts [hunk 3]
```diff
@@ -231,14 +231,38 @@ export const runAction = async (space): Promise<void> => {
...
+
+  // If CDA token is created and we want to purge ephemeral env upon close/merge
+  // then ephemeral token should be deleted as well
+  if ( 
+    DELETE_FEATURE && 
+    CREATE_CDA_TOKEN && 
+    githubAction === "closed" 
+  ) {
+    const { items: keys } = await space.getApiKeys();
+    keys.map((key) => {
```
[the remark this answers is not in the document]
Comment (slash3b) [reply src/action.ts]: `keys.find` worked like a charm, and there is no need to loop over all the keys.
excellent suggestion, thank you 🙌

File: src/action.ts (hunk 2 above)
Comment (slash3b) [reply src/action.ts]: not sure, but it definitely works like this fine 
can I leave it like it is?

Comment (kayakyakr) [reply src/action.ts]: So, it executes fine without the await, but if it's an async function (best I can tell, it is), then the catch will never catch the response and you will have no indication that there was an error. Fire & forget vs fire & wait. 

If your intent is to fire & forget, and therefore ignore any failures, then would want to indicate that in comments.

Comment (slash3b) [reply src/action.ts]: I definitely do not want to "fire&forget", but rather catch error/exception if any.
seems like `.then().catch()` contraption is the way to go.

Comment (slash3b) [reply src/action.ts]: just `await`-ing on delete() 👍

File: src/action.ts [hunk 4]
```diff
@@ -231,14 +231,42 @@ export const runAction = async (space): Promise<void> => {
...
+    CREATE_CDA_TOKEN && 
+    githubAction === "closed" 
+  ) {
+    Logger.verbose(`debug: attempting to delete ${tokenKeyName}`)
+
+    const { items: keys } = await space.getApiKeys();
+    const k = keys.find(key => key.name === tokenKeyName)
+
+    if (k === undefined) {
+        Logger.warn(`could not find ephemeral toke ${tokenKeyName}, possibly it was deleted manually`);
```
Comment (kayakyakr) [inline src/action.ts:250]:
```suggestion
        Logger.warn(`could not find ephemeral token ${tokenKeyName}, possibly it was deleted manually`);
```

File: src/action.ts [hunk 5]
```diff
@@ -231,14 +231,42 @@ export const runAction = async (space): Promise<void> => {
...
+    } else {
+      try {
+        await k.delete();
+        Logger.success(`removed ephemeral token ${tokenKeyName}`);
+      } catch(error) {
+        Logger.error("Unable to delete ephemeral token");
+        Logger.error(error);
```
Comment (kayakyakr) [inline src/action.ts:257]: should the additional error messaging `warn` or `info` rather than error so it gets shown or not based on a `--verbose` flag?

**Review by kayakyakr — DISMISSED** [review]
Looks great. Thought up 2 more minor things, but those don't block approval.

**Review by kayakyakr — APPROVED** [review]

**Pull request merged**